streamingpro-mlsql/src/main/java/streaming/dsl/mmlib/algs/SQLDownloadExt.scala

```diff
@@ -123,9 +124,15 @@ class SQLDownloadExt(override val uid: String) extends SQLAlg with DslTool with
         if (tarIS.canReadEntryData(entry)) {
           if (!entry.isDirectory) {
             val dir = entry.getName.split("/").filterNot(f => f.isEmpty).dropRight(1).mkString("/")
-            downloadResultRes += DownloadResult(PathFun(originalTo).add(dir).add(entry.getName.split("/").last).toPath)
+            val fileName = entry.getName.split("/").last
+            downloadResultRes += DownloadResult(PathFun(originalTo).add(dir).add(fileName).toPath)
             logInfo(format(s"extracting ${downloadResultRes.last.hdfsPath}"))
-            HDFSOperatorV2.saveStream($(to) + "/" + dir, entry.getName.split("/").last, tarIS)
+            if (DecompressUtil.isArchiveFile(fileName)){
```

Comment:
So this PR is designed that we can decompress the compressed file in tar file?

Reply:
Yes, also support decompress tar, tar.gz, gz, zip files

Reply:
> So this PR is designed that we can decompress the compressed file in tar file?

Yes, also support decompress tar, tar.gz, gz, zip files

Reply:
The reason we use tar format is that it's not efficient when the user want to upload  thousands of files. 
Two much files are slow and more easy to fail. So if people upload tar file , we will automatically extract it.
However, there is no need to extract zip file in tar file since the user may need the zip file in hdfs/object store.

So you can change this PR to support extract uploaded file with tar, tar.gz format instead of the files in it.